### ISSUE TITLE
Create electronic schematic for robot

### ISSUE THREAD

**Issue by vyosypenko** [ros-ukraine/leobot#124]
Schematic will be created in free EDA KiCad.

**Comment by LyubomyrD** [comment]
Great choice!

**Comment by vyosypenko** [comment]
lib with STM32F4 board headers: http://smisioto.no-ip.org/elettronica/kicad/kicad-en.htm

Symbol and footprint
https://www.snapeda.com/parts/STM32F4DISCOVERY/STMicroelectronics/view-part/

Headers 3D model:
https://grabcad.com/library/stm32f4discovery-socket-for-kicad-1

**Comment by vyosypenko** [comment]
Added 3D model for the footprint in KiCad. As for me looks nice :)
![image](https://user-images.githubusercontent.com/7555013/40049158-9fc3c0d0-583c-11e8-9aa5-6a3ef9ebcf6f.png)

**Comment by vyosypenko** [comment]
![image](https://user-images.githubusercontent.com/7555013/40292241-af1503a6-5cd2-11e8-958f-14c0fb2cfb64.png)

**Comment by vyosypenko** [comment]
KiCad project is located: https://github.com/ros-ukraine/leobot/tree/F%23124_MCU_Board_Schematic/docs/design/hardware/MCU_Board

**Comment by vyosypenko** [comment]
1. Connection diagram:
![image](https://user-images.githubusercontent.com/7555013/41605452-1a00f980-73ea-11e8-83c1-3c75962ddc53.png)

2. Preliminary schematic:
![image](https://user-images.githubusercontent.com/7555013/41605534-4a5d8c38-73ea-11e8-9887-98c66ac9984c.png)

**Comment by LyubomyrD** [comment]
@nemo8 great schematics!
It seems like we need both InA and InB for motors drive shield.
I guess that we have to consider powering drive shields from 5V

**Comment by vyosypenko** [comment]
@LyubomyrD , 

thanks! Good catch!

I have made some mistakes with motor driver signals. 

You are right:
1.  5v should be connected to shields.
2.  And there are no  "InA and InB" lines on the current schematic. 

I will update schematic.

If You wish you can open schematic in Kicad, the current project is located here:
https://github.com/ros-ukraine/leobot/tree/firmware/docs/design/hardware/MCU_Board

**Comment by vyosypenko** [comment]
Hi @LyubomyrD ,
according to the motor driver schematic VCC is used only for two signal pull-up (we don't use these pins in our design), so it doesn't really matter 5V or 3.3V:

![image](https://user-images.githubusercontent.com/7555013/41759629-1ae7c1de-75f8-11e8-9feb-5f86b6916895.png)

connected motor driver signals INA, INB to the stm32 board connector:
![image](https://user-images.githubusercontent.com/7555013/41759692-653661be-75f8-11e8-93e4-0c4f46043817.png)

**Comment by LyubomyrD** [comment]
@nemo8 as far as I understand, pull-ups are for the motor_enable pin. I guess, that even if we do not use it now, we should add the usage to the schematics for safety reasons and possible power usage optimization (when robot is idle, there is no need for motors to be ebanled). @AndriyPt , please, confirm.

**Comment by vyosypenko** [comment]
@LyubomyrD yes, these pins are for motor enable. For this pins 5V and 3.3V are fine.

Motor is disabled in the next cases 
1. INA and INB = 0
2. INA and INB = 1
3. PMW = 0



![image](https://user-images.githubusercontent.com/7555013/41776258-ae8285be-762f-11e8-924e-09a6674759d7.png)

To disable motor using  ENA/DIAGA, ENB/DIAGB  just need to take off supply volatage  (and pininternally will be connected to GND). These pins are dedicated for monitoring fault detection. If we want to disable motor via ENA/DIAGA, ENB/DIAGB pins we need to add the transistor switch for 5V line. 
![image](https://user-images.githubusercontent.com/7555013/41776318-f29fe9b2-762f-11e8-87ce-f3731d1f5a3f.png)
